api/tests/test_user.py

```diff
@@ -1,57 +1,232 @@
         """
-        response = self.client.get(reverse("api:logged_user"))
+        user = self.login()
+        response = self.get(self.url())
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
-        body = json.loads(response.content.decode())
-        self.assertEqual(body.get("username"), authenticate.user.username)
-        self.assertEqual(body.get("email"), authenticate.user.email)
-        self.assertEqual(body.get("firstName"), authenticate.user.first_name)
-        self.assertEqual(body.get("lastName"), authenticate.user.last_name)
-        self.assertEqual(body.get("roles"), [])
-        self.assertIn("id", body)
-
-    @authenticate
+        self.assertEqual(response.data.get("id"), user.id)
+        self.assertEqual(response.data.get("username"), user.username)
+        self.assertEqual(response.data.get("email"), user.email)
+        self.assertEqual(response.data.get("first_name"), user.first_name)
+        self.assertEqual(response.data.get("last_name"), user.last_name)
+        self.assertEqual(response.data.get("roles"), [])
+        self.assertIn("id", response.data)
+
     def test_authenticated_logged_user_call_with_roles(self):
         """Ensure that roles are added to the JSON representation of the user"""
 
         def _get_role_names(resp):
             return {role["name"] for role in resp.data["roles"]}
 
+        user = self.login()
+
         # Without role
-        response = self.client.get(reverse("api:logged_user"))
+        response = self.get(self.url())
         self.assertEqual(_get_role_names(response), set())
 
         # With two roles
-        DeclarantFactory(user=authenticate.user)
-        CompanySupervisorFactory(user=authenticate.user)
+        DeclarantFactory(user=user)
+        CompanySupervisorFactory(user=user)
 
-        response = self.client.get(reverse("api:logged_user"))
+        response = self.get(self.url())
         self.assertEqual(_get_role_names(response), {"Declarant", "CompanySupervisor"})
 
 
-class TestGenerateUsername(APITestCase):
+class TestSignup(ProjectAPITestCase):
+    viewname = "signup"
+
+    def setUp(self):
+        self.user_data = dict(
+            last_name="Cook", first_name="Tim", email="tim.cook@exemple.com", username="tcook", password="azerty123$"
+        )
+
+    def test_signup_ok(self):
+        response = self.post(self.url(), self.user_data)
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+        self.assertIn("user_id", response.data)
+        new_user = User.objects.get(id=response.data["user_id"])
+        self.assertEqual(new_user.email, self.user_data["email"])
+        self.assertTrue(new_user.is_active)
+        self.assertFalse(new_user.is_verified)
+        self.assertFalse(new_user.is_superuser)
+        self.assertFalse(new_user.is_staff)
+        self.assertEqual(new_user.roles, [])
+
+    def test_signup_with_existing_email(self):
+        email = "eren@mikasa.com"
```

Comment:
Pour les adresses email et les domaines fake on peut utiliser _example.com_, qui fait partie des [domaines réservés](https://en.wikipedia.org/wiki/Example.com) par la IANA pour faire des tests ou servir de placeholders.
```suggestion
        email = "eren@example.com"
```